tss-esapi-sys/build.rs

```diff
@@ -59,19 +69,158 @@ pub mod tpm2_tss {
+                    let windows_sdk = ip_pb.join("10.0.17134.0");
+                    clang_args.push(format!("-I{}", windows_sdk.join("ucrt").display()));
+                    clang_args.push(format!("-I{}", windows_sdk.join("um").display()));
+                    clang_args.push(format!("-I{}", windows_sdk.join("shared").display()));
+                    return Some(clang_args);
+                }
+                else {
+                    return None;
+                }
+            }
+        }
+
+        #[cfg(feature = "bundled")]
+        /// Fetch the given source repo using git
+        fn fetch_source(
+            dest_path: impl AsRef<std::path::Path>,
+            name: &str,
+            repo: &str,
+            branch: &str,
+        ) -> std::path::PathBuf {
+            let parent_path = dest_path.as_ref();
+            let repo_path = parent_path.join(name);
+            if !repo_path.join("Makefile.am").exists() {
+                let output = std::process::Command::new("git")
+                    .args(["clone", repo, "--depth", "1", "--branch", branch])
+                    .current_dir(parent_path)
+                    .output()
+                    .expect(&format!("git clone for {} failed", name));
+                let status = output.status;
+                if !status.success() {
+                    panic!(
+                        "git clone for {} returned failure status {}:\n{:?}",
+                        name, status, output
+                    );
+                }
+            }
+
+            repo_path
+        }
+
+        #[cfg(feature = "bundled")]
+        fn compile_with_autotools(p: PathBuf) -> PathBuf {
+            let output1 = std::process::Command::new("./bootstrap")
+                .current_dir(&p)
+                .output()
+                .expect("bootstrap script failed");
+            let status = output1.status;
+            assert!(
+                status.success(),
+                "bootstrap script failed with {status}:\n{output1:?}"
+            );
+
+            let mut config = autotools::Config::new(p);
+            config.fast_build(true).reconf("-ivf").build()
+        }
+
+        #[cfg(feature = "bundled")]
+        /// Uses a bundled build for an installation
+        pub fn bundled() -> Self {
+            use std::io::Write;
+            let out_path = std::env::var("OUT_DIR").expect("No output directory given");
+            let source_path = Self::fetch_source(
+                out_path,
+                "tpm2-tss",
+                "https://github.com/tpm2-software/tpm2-tss.git",
+                MINIMUM_VERSION,
+            );
+            let version_file_name = source_path.join("VERSION");
+            let mut version_file = std::fs::File::create(version_file_name)
+                .expect("Unable to create version file for tpm2-tss");
+            write!(version_file, "{}", MINIMUM_VERSION);
```

Comment:
You need to handle the returned result of write! with an `unwrap_or_else(|e| panic!("Failed to write version file: {e}"))`